src/routes/hotels.php

```diff
@@ -6,28 +6,236 @@
+
     try {
-        // Get DB Object
         $db = new db();
-        // Connect
+
         $db = $db->connect();
-        $stmt = $db->query($sql);
+        $stmt = $db->query($query);
+
         $hotels = $stmt->fetchAll(PDO::FETCH_OBJ);
         $db = null;
 
         $response->write(json_encode($hotels));
         return $response->withStatus(200);
     } catch (PDOException $e) {
+        $db = null;
+
+        $response->write($e);
+        return $response->withStatus(500);
+    }
+});
+
+$app->post('/api/hotels/create', function (Request $request, Response $response) {
+
+    $parsedBody = $request->getParsedBody();
+
+    $createAddressQuery = "INSERT INTO address (country,province,city, street, postal_code)" .
+        "VALUES (:country, :province, :city, :street, :postal_code)";
+    $createHotelQuery = "INSERT INTO hotel (brand_name, branch_name, property_class, address_id, description, overall_rating, phone_number)" .
+        " VALUES (:brand_name,:branch_name,:property_class,:address_id,:description,:overall_rating, :phone_number)";
+
+    $db = new db();
+    $db = $db->connect();
+
+    try {
+        $db->beginTransaction();
+
+        $stmt = $db->prepare($createAddressQuery);
+
+        $stmt->bindParam(":country", $parsedBody["country"]);
+        $stmt->bindParam(":city", $parsedBody["city"]);
+        $stmt->bindParam(":province", $parsedBody["province"]);
+        $stmt->bindParam(":street", $parsedBody["street"]);
+        $stmt->bindParam(":postal_code", $parsedBody["postal_code"]);
+
+        if ($stmt->execute()) {
+            $insertedId = $db->lastInsertId();
+
+            $stmt = $db->prepare($createHotelQuery);
+
+            $stmt->bindParam(":brand_name", $parsedBody["brand_name"]);
+            $stmt->bindParam(":branch_name", $parsedBody["branch_name"]);
+            $stmt->bindParam(":property_class", $parsedBody["property_class"]);
+            $stmt->bindParam(":address_id", $insertedId);
+            $stmt->bindParam(":description", $parsedBody["description"]);
+            $stmt->bindParam(":overall_rating", $parsedBody["overall_rating"]);
+            $stmt->bindParam(":phone_number", $parsedBody["phone_number"]);
+
+            if ($stmt->execute()) {
+                $insertedId = $db->lastInsertId();
+                $db->commit();
+                $db = null;
+
+                $responseArray = array();
+                $responseArray["id"] = $insertedId;
+
+                return $response->write(json_encode($responseArray))->withStatus(200);
+            } else {
+                $db = null;
+
+                $response->write('{"error": {"text": "failed create on hotel"}}');
+                return $response->withStatus(500);
+            }
+        } else {
+            $db = null;
+
+            $response->write('{"error": {"text": "failed create on address"}}');
+            return $response->withStatus(500);
```

Comment:
you can only set the status code `$response->withStatus(500)` and return at the end of the function